### PR TITLE
Attempt to fix missed session end

### PR DESCRIPTION
**Pull request by by-dam**
When session props are updated after sending sensor data it uses
the props that existed when it started to send the data.

If the end date is set while sending data it is wiped out at this time.

### CODE REVIEW

File: MuvrKit/MKWatchConnectivity.swift [hunk 1]
```diff
@@ -249,20 +249,17 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
 
             // update the number of recorded samples
             let readFromDate = NSDate(timeIntervalSince1970: sensorData.end)
-            let updatedProps = props.with(accelerometerEnd: readFromDate)
-            self.sessions[session] = updatedProps
+            if let p = self.sessions[session] {
```
Comment (janm399) [inline MuvrKit/MKWatchConnectivity.swift]: Can this actually not be true? We got the session a few lines earlier.

Comment (by-dam) [inline MuvrKit/MKWatchConnectivity.swift]: Correct, this one is probably fine.
I think the problem lies in the transfer data callback

File: MuvrKit/MKWatchConnectivity.swift [hunk 2]
```diff
@@ -249,20 +249,17 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
...
             
             // transfer what we have so far
-            transferSensorDataBatch(sensorData, session: session, props: updatedProps) {
-                // REVIEW: does this fix the sessions problem?
-                if self.sessions[session] == nil {
-                    NSLog("Session \(session) already removed.")
-                    dispatch_async(self.transferQueue, processFirstSession)
-                } else {
-                    // set the expected range of samples on the next call
-                    let finalProps = updatedProps.with(accelerometerStart: readFromDate)
```
Comment (by-dam) [inline MuvrKit/MKWatchConnectivity.swift]: updatedProps was read before sending the data and might be out of date at this point.

Comment (janm399) [inline MuvrKit/MKWatchConnectivity.swift]: I don't think so: the ``onDone`` block gets called after the file transfer.

Comment (by-dam) [inline MuvrKit/MKWatchConnectivity.swift]: exactly and updatedProps comes from props that was read before the file transfer.
so if the session ends during this time the end date is lost.

**Pull request closed without merging**